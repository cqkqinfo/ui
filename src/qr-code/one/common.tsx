import { ImageProps } from 'remax/one';

export default interface QrCodeProps extends ImageProps {
  /**
   * 二维码内容
   */
  content: string;
  /**
   * 二维码码的颜色
   */
  darkColor?: string;
  /**
   * 二维码背景颜色
   */
  lightColor?: string;
  /**
   * 设置二维码回调
   */
  onSetSrc?: (src: string) => void;
  /**
   * 长按保存
   */
  longTapSave?: boolean;
  /**
   * 点击显示弹窗
   */
  showModal?: boolean;
  /**
   * 弹窗标题
   */
  modalTitle?: React.ReactNode;
}
