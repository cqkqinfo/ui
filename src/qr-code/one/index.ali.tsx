/* eslint-disable @typescript-eslint/ban-ts-comment */
import React, { useEffect, useMemo } from 'react';
import { Image } from 'remax/ali';
import QrCodeProps from './common';
// @ts-ignore
import QR from 'qrcode-base64';

export default ({
  content,
  lightColor: _,
  darkColor: __,
  className,
  onSetSrc,
  longTapSave,
  ...restProps
}: QrCodeProps) => {
  const src = useMemo(
    () =>
      QR.drawImg(content, {
        typeNumber: 4,
        errorCorrectLevel: 'L',
        size: 500,
      }),
    [content],
  );
  useEffect(() => {
    onSetSrc?.(src);
  }, [onSetSrc, src]);
  return <Image className={className} src={src} {...(restProps as any)} />;
};
