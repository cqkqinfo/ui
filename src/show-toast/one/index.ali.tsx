import { showToast } from 'remax/ali';

export interface Options {
  title: string;
  icon?: 'success' | 'none';
  mask?: boolean;
  duration?: number;
}

export default (options: Options) =>
  showToast({
    content: options.title,
    type: options.icon,
    duration: options.duration,
  });
