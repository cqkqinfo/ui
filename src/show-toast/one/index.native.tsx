import { Toast } from '@ant-design/react-native';
import { Options } from './index.wechat';

export default ({
  icon = 'success',
  title,
  duration = 1000,
  mask = false,
}: Options) =>
  new Promise(resolve => {
    if (icon === 'none') {
      Toast.fail(title, duration / 1000, () => resolve(''), mask);
    } else {
      Toast.success(title, duration / 1000, () => resolve(''), mask);
    }
  });
