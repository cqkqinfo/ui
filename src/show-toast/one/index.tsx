import { Toast } from 'antd-mobile';
import { Options } from './index.wechat';

export default ({ icon = 'success', title, duration, mask = false }: Options) =>
  new Promise(resolve => {
    Toast.show({
      icon: icon === 'none' ? undefined : 'success',
      content: title,
      duration,
      maskClickable: mask,
      afterClose: () => resolve(''),
    });
  });
