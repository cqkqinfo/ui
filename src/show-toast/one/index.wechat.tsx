import { showToast } from 'remax/wechat';

export interface Options {
  title: string;
  icon?: 'success' | 'none';
  mask?: boolean;
  duration?: number;
}

export default (options: Options) => showToast(options);
