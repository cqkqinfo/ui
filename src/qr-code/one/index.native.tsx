/* eslint-disable @typescript-eslint/ban-ts-comment */
import React from 'react';
// @ts-ignore
import QrCodeProps from './common';
import QRCode from 'react-native-qrcode-svg';
import { View } from 'remax/one';
import useViewLayout from '../../use-view-layout';

export default ({
  content,
  darkColor,
  lightColor,
  onSetSrc,
  longTapSave,
  style,
  ...restProps
}: QrCodeProps) => {
  const { width, ...arg } = useViewLayout();
  return (
    <View style={style} {...restProps} {...arg}>
      {!!width && <QRCode value={content} size={width} {...restProps} />}
    </View>
  );
};
