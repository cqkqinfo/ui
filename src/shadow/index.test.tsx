import React from 'react';
import Shadow from './index';
import { render } from '@testing-library/react';
import { View } from 'remax/one';

describe('阴影', () => {
  it('渲染', () => {
    const { baseElement } = render(
      <Shadow>
        <View>233</View>
      </Shadow>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('卡片模式', () => {
    const { baseElement } = render(
      <Shadow card>
        <View>233</View>
      </Shadow>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
