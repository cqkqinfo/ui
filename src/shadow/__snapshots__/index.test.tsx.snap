// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`阴影 卡片模式 1`] = `
<body>
  <div>
    <div
      style="box-sizing: border-box; background-color: rgb(255, 255, 255); border-radius: 10px; padding: 0px 10px; box-shadow: 0 0 10px rgba(39,128,217, 0.15);"
    >
      233
    </div>
  </div>
</body>
`;

exports[`阴影 渲染 1`] = `
<body>
  <div>
    <div
      style="box-shadow: 0 0 10px rgba(39,128,217, 0.15);"
    >
      233
    </div>
  </div>
</body>
`;
